src/v/utils/BUILD

```diff
@@ -0,0 +1,364 @@
+    deps = [
+    ],
+)
+
+redpanda_cc_library(
+    name = "to_string",
+    hdrs = [
+        "to_string.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+    ],
+)
+
+redpanda_cc_library(
+    name = "object_pool",
+    hdrs = [
+        "object_pool.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+    ],
+)
+
+redpanda_cc_library(
+    name = "human",
+    srcs = [
+        "human.cc",
+    ],
+    hdrs = [
+        "human.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_library(
+    name = "unresolved_address",
+    hdrs = [
+        "unresolved_address.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/serde",
+    ],
+)
+
+redpanda_cc_library(
+    name = "hdr_hist",
+    srcs = [
+        "hdr_hist.cc",
+    ],
+    hdrs = [
+        "hdr_hist.h",
+    ],
+    exclude_layering_check = True,
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":human",
+        ":static_deleter_fn",
+        "//src/v/base",
+        "@boost//:intrusive",
+        "@fmt",
+        "@hdrhistogram",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_library(
+    name = "log_hist",
+    srcs = [
+        "log_hist.cc",
+    ],
+    hdrs = [
+        "log_hist.h",
+    ],
+    exclude_layering_check = True,
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/container:fragmented_vector",
+        "//src/v/reflection:adl",
+    ],
+)
+
+redpanda_cc_library(
+    name = "uuid",
+    srcs = [
+        "uuid.cc",
+    ],
+    hdrs = [
+        "uuid.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+        "//src/v/bytes",
+        "//src/v/bytes:iobuf",
+        "@abseil-cpp//absl/hash",
+        "@boost//:uuid",
+        "@fmt",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_library(
+    name = "xid",
+    srcs = [
+        "xid.cc",
+    ],
+    hdrs = [
+        "xid.h",
+    ],
+    include_prefix = "utils",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/base",
+        "//src/v/bytes:iobuf_parser",
+        "//src/v/serde",
+        "//src/v/serde:array",
+        "//src/v/ssx:sformat",
+        "@fmt",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_btest_no_seastar(
+    name = "tristate_test",
+    timeout = "short",
+    srcs = [
+        "tests/tristate_test.cc",
+    ],
+    defines = [
+        "BOOST_TEST_MODULE=tristate",
+    ],
+    deps = [
+        ":tristate",
+    ],
+)
+
+redpanda_cc_btest_no_seastar(
+    name = "named_type_test",
+    timeout = "short",
+    srcs = [
+        "tests/named_type_tests.cc",
```

Comment:
Throughout the whole PR - why not have a BUILD file in the tests directory?

Reply:
it doesn't matter to me. it was slightly less typing in vim completion, but the some of the files are getting large. wdyt? i'm happy to split them up!

Reply:
I am used to BUILD file per directory and you generally don't reference things outside that directory. But then again Google put tests in the same directory as sources, so the size of stuff is fine IMO

Reply:
I think BUILD per directory is cleaner, and we can decide if we want separate directories as a separate decision. Might as well make the change now while there are relatively few BUILD files.

Reply:
Agreed. FWIW with bazel I suspect it will feel more natural to break up and have finer grained packages (and targets). With that pattern having tests in the same directory as sources won't feel so cluttered. I am also fine with keeping tests separate too

Reply:
Fixed